### PR TITLE
Do not report references to object type from Class objects

### PR DESCRIPTION
**Pull request by pdbain-ibm**
Per JVMTI specification for FollowReferences(), do not report the class of
Class objects.

Signed-off-by: Peter Bain <peter_bain@ca.ibm.com>

### CODE REVIEW

**Comment by pdbain-ibm** [comment]
@DanHeidinga or @gacholio would you kindly do an initial review while I regression test the change?

**Comment by DanHeidinga** [comment]
@pdbain-ibm Has the expected behaviour changed between different JDK releases?  Which version of the JVMTI spec describes this?

**Comment by pdbain-ibm** [comment]
I am referring to Version 1.2 of the JVMTI specification, so I believe this behaviour was intended from the beginning, though never tested.  We now have a test which checks the kinds of references reported.

File: runtime/jvmti/jvmtiHeap.c [hunk 1]
```diff
@@ -1137,7 +1137,20 @@ mapEventType(J9JVMTIHeapData * data, IDATA type, jint index, j9object_t referrer
             event->type = J9JVMTI_HEAP_EVENT_OBJECT;
```
Comment (gacholio) [inline runtime/jvmti/jvmtiHeap.c]: While you're here, please fix this line to use tabs to indent.

**Comment by gacholio** [comment]
Change looks reasonable to me.  I'll do a final review once the testing has completed.

File: runtime/jvmti/jvmtiHeap.c (hunk 1 above)
Comment (pdbain-ibm) [reply runtime/jvmti/jvmtiHeap.c]: Done.

**Comment by pdbain-ibm** [comment]
Ready for merging.  Please run Jenkins test builds.  Thank you.

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux

**Pull request merged**